test/integration/server.h

```diff
@@ -298,11 +352,31 @@ class IntegrationTestServer : public Logger::Loggable<Logger::Id::testing>,
              bool defer_listener_finalization, ProcessObjectOptRef process_object,
              Server::FieldValidationConfig validation_config, uint32_t concurrency);
 
+  using AwaitData = std::tuple<Stats::NotifyingAllocatorImpl*, const std::string*, uint64_t>;
   void waitForCounterEq(const std::string& name, uint64_t value) override {
-    TestUtility::waitForCounterEq(statStore(), name, value, time_system_);
+    while (statsAllocator().getCounter(name) == nullptr ||
+           statsAllocator().getCounter(name)->value() != value) {
+      AwaitData data = std::make_tuple(&statsAllocator(), &name, value);
+      statsAllocator().mutex().Await(absl::Condition(+[](AwaitData* data) {
```

Comment:
oh yeah I have an outstanding PR which provides a modern wrapper for Await to hide this messiness.

Reply:
Ok if I revetr this while that lands?  it was a WIP and again I think it decreases readability.